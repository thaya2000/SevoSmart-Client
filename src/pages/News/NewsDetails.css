/* NewsPage.css */
  
  .news-page h1 {
    font-size: 2.5rem;
    font-weight: bold;
  }
  
  .news-feed {
    max-width: 800px;
    margin: 0 auto;
  }
  
  .feed-items {
    display: grid;
    grid-template-columns: repeat(4, 1fr); /* Ensure three columns */
    gap: 20px;
  }
  
  .feed-item {
    background-color: #f9f9f9;
    border-radius: 5px;
    box-shadow: 0 0 10px rgba(0, 0, 0, 0.1);
    overflow: hidden;
    display: flex;
    flex-direction: column;
    padding: 10px;
    max-width: 300px; /* Limit the width of each item */
    margin: auto; /* Center the items horizontally */
  }
  
  
  .feed-item img {
    width: 100%;
    height: auto;
    border-radius: 5px;
  }
  
  .feed-item h2 {
    margin: 5px;
    font-weight: bold;
    font-size: 1.2rem;
  }
  
  .feed-item p {
    margin: 5px;
  }

    .feed-item button {
      margin: auto;
      padding: 5px 8px; /* Reduce padding */
      background-color: #007bff;
      color: #fff;
      border: none;
      border-radius: 5px;
      cursor: pointer;
      width: 100px; /* Set a specific width */
      
    }
    
    .feed-item button:hover {
      background-color: #0056b3;
    }
    