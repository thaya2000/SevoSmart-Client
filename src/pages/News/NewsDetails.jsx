// NewsDetails.jsx

import React from 'react';
import { useParams } from 'react-router-dom';
import './NewsDetails.css';
import image1 from '../../Images/NewsImages/im1.jpg';
import image2 from '../../Images/NewsImages/im2.jpg';
import image3 from '../../Images/NewsImages/im3.jpg';
import image4 from '../../Images/NewsImages/im4.jpg';
import image5 from '../../Images/NewsImages/im5.jpg';
import image6 from '../../Images/NewsImages/im6.jpg';
import image7 from '../../Images/NewsImages/im7.jpg';
import image8 from '../../Images/NewsImages/im1.jpg';
import image9 from '../../Images/NewsImages/im2.jpg';
import image10 from '../../Images/NewsImages/im3.jpg';
import image11 from '../../Images/NewsImages/im4.jpg';
import image12 from '../../Images/NewsImages/im5.jpg';

const newsData = [
  { id: 1, title: "Welcome to our news channel in sevosmart", image: image1, content: "This is a sample paragraph. Read more to view the full news." },
  { id: 2, title: "Welcome to our news channel in sevosmart", image: image2, content: "This is a sample paragraph. Read more to view the full news." },
  { id: 3, title: "Welcome to our news channel in sevosmart", image: image3, content: "This is a sample paragraph. Read more to view the full news." },
  { id: 4, title: "Welcome to our news channel in sevosmart", image: image4, content: "This is a sample paragraph. Read more to view the full news." },
  { id: 5, title: "Welcome to our news channel in sevosmart", image: image5, content: "This is a sample paragraph. Read more to view the full news." },
  { id: 6, title: "Welcome to our news channel in sevosmart", image: image6, content: "This is a sample paragraph. Read more to view the full news." },
  { id: 7, title: "Welcome to our news channel in sevosmart", image: image7, content: "This is a sample paragraph. Read more to view the full news." },
  { id: 8, title: "Welcome to our news channel in sevosmart", image: image8, content: "This is a sample paragraph. Read more to view the full news." },
  { id: 9, title: "Welcome to our news channel in sevosmart", image: image9, content: "This is a sample paragraph. Read more to view the full news." },
  { id: 10, title: "Welcome to our news channel in sevosmart", image: image10, content: "This is a sample paragraph. Read more to view the full news." },
  { id: 11, title: "Welcome to our news channel in sevosmart", image: image11, content: "This is a sample paragraph. Read more to view the full news." },
  { id: 12, title: "Welcome to our news channel in sevosmart", image: image12, content: "This is a sample paragraph. Read more to view the full news." },
];

const NewsDetails = () => {
  const { id } = useParams();
  const news = newsData.find(item => item.id === parseInt(id));

  if (!news) {
    return <div className="news-details">News not found</div>;
  }

  return (
    <div className="news-details">
      <h1>{news.title}</h1>
      <img src={news.image} alt={news.title} className="large-image" />
      <p>{news.content}</p>
    </div>
  );
};

export default NewsDetails;
