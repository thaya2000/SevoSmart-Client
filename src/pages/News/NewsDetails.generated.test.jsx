import { describe, it, expect, vi } from 'vitest';
import { render } from '@testing-library/react';
import NewsDetails from './NewsDetails';

// Mocking necessary modules and assets
vi.mock('react-router-dom', () => ({
  useParams: vi.fn(() => ({ newsId: '123' })) // Example of how to mock useParams if needed
}));
vi.mock('./NewsDetails.css');
vi.mock('../../Images/NewsImages/im1.jpg');
vi.mock('../../Images/NewsImages/im2.jpg');
vi.mock('../../Images/NewsImages/im3.jpg');
vi.mock('../../Images/NewsImages/im4.jpg');
vi.mock('../../Images/NewsImages/im5.jpg');
vi.mock('../../Images/NewsImages/im6.jpg');
vi.mock('../../Images/NewsImages/im7.jpg');

describe('<NewsDetails>', () => {
  it('should render component', () => {
    const { asFragment } = render(<NewsDetails />);
    expect(asFragment()).toMatchSnapshot();
  });
});
