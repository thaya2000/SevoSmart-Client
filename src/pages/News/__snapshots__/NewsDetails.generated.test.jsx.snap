// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<NewsDetails> > should render component 1`] = `
<DocumentFragment>
  <div
    class="news-details"
  >
    News not found
  </div>
</DocumentFragment>
`;
